web-app/HeimScripts/heatmap/summary.R

```diff
@@ -151,12 +257,17 @@ extract_measurements <- function(datasets)
 produce_summary_stats <- function(measurement_tables, phase)
 {
   # construct data.frame to store the results from the summary statistics in
-  result_table <- as.data.frame(matrix(NA, length(measurement_tables),12, 
+  result_table <- as.data.frame(matrix(NA, length(measurement_tables),13, 
                                        dimnames = list(names(measurement_tables), 
-                                                       c("variableLabel","node","subset","totalNumberOfValuesIncludingMissing", "numberOfMissingValues", "min","max","mean", "standardDeviation", "q1","median","q3"))))
+                                                       c("variableLabel","node","subset","totalNumberOfValuesIncludingMissing", "numberOfMissingValues", "numberOfSamples","min","max","mean", "standardDeviation", "q1","median","q3"))))
+  
   # add information about node and subset identifiers
-  result_table$subset <- gsub(".*_","", rownames(result_table))
-  result_table$node <- gsub("_.*","", rownames(result_table))
+  result_table$subset <- gsub(".*_","", rownames(result_table)) #take everything after _
```

Comment:
No, it doesn't. For loaded_variables the labels are used which are always n<numerical node id>_s<numerical subset id>. For preprocessed the variable preprocessed is turned into a list by get_input_data, with list("preprocessed" = preprocessed). Later this is split in "split_on_subsets" and then preprocessed contains something like list("preprocessed_s1"  = <some portion of the table> , "preprocessed_s2" = <remainder of table>), it is these two labels ("preprocessed_s<num>") that are split,  so this portion of the code does not have to deal with subject names. In "split_on_subsets" strict matching is used to extract the subsets.